### PR TITLE
Extract nagios types into their own module

### PR DESCRIPTION
**Pull request by jhelwig**

### CODE REVIEW

File: .sync.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - gem: beaker-abs
+        from_env: BEAKER_ABS_VERSION
+        version: '~> 0.1'
+      - gem: beaker-pe
+      - gem: beaker-hostgenerator
+        from_env: BEAKER_HOSTGENERATOR_VERSION
+      - gem: beaker-rspec
+        from_env: BEAKER_RSPEC_VERSION
+    ':development':
+      - gem: puppet-blacksmith
```
Comment (melissa) [inline .sync.yml]: pdk 1.5.0 should have this now, so we shouldn't have to add it here anymore, same with line 30 requiring the tasks

File: Gemfile [hunk 2]
```diff
@@ -28,84 +28,48 @@ group :development do
   gem "fast_gettext",                                  require: false if Gem::Version.new(RUBY_VERSION.dup) >= Gem::Version.new('2.1.0')
   gem "json_pure", '<= 2.0.1',                         require: false if Gem::Version.new(RUBY_VERSION.dup) < Gem::Version.new('2.0.0')
   gem "json", '= 1.8.1',                               require: false if Gem::Version.new(RUBY_VERSION.dup) == Gem::Version.new('2.1.9')
+  gem "json", '<= 2.0.4',                              require: false if Gem::Version.new(RUBY_VERSION.dup) == Gem::Version.new('2.4.4')
   gem "puppet-module-posix-default-r#{minor_version}", require: false, platforms: [:ruby]
   gem "puppet-module-posix-dev-r#{minor_version}",     require: false, platforms: [:ruby]
   gem "puppet-module-win-default-r#{minor_version}",   require: false, platforms: [:mswin, :mingw, :x64_mingw]
   gem "puppet-module-win-dev-r#{minor_version}",       require: false, platforms: [:mswin, :mingw, :x64_mingw]
+  gem "puppet-blacksmith", '~> 3.4',                   require: false, platforms: [:ruby]
+  gem "puppet-blacksmith", '~> 3.4',                   require: false
```
Comment (melissa) [inline Gemfile]: yeah, it gets listed twice when we have it in `.sync.yml`

File: Rakefile [hunk 3]
```diff
@@ -1,2 +1,7 @@
 require 'puppetlabs_spec_helper/rake_tasks'
 require 'puppet-syntax/tasks/puppet-syntax'
+require 'puppet_blacksmith/rake_tasks' if Bundler.rubygems.find_name('puppet-blacksmith').any?
+require 'puppet_blacksmith/rake_tasks'
```
Comment (melissa) [inline Rakefile]: listed twice, you should remove this from `.sync.yml`

**Review by melissa — APPROVED** [review]
It looks like you're not running into the same travis failures that I was dealing with in zfs. How did you get around that here?

I only have small nits, otherwise I think this looks good!

**Comment by jhelwig** [comment]
Removed the extra `puppet-blacksmith` references from `.sync.yml`, and ran `pdk update` to regenerate the Gemfile & Rakefile.

**Pull request merged**